### ISSUE TITLE
Overlapping subjoined Myanmar consonants between narrow base and medial ra

### ISSUE THREAD

**Issue by dscorbett** [googlefonts/noto-fonts#1711]
### Font
NotoSansMyanmar-Regular.otf
NotoSerifMyanmar-Regular.ttf

### Where the font came from, and when
Site: https://github.com/googlefonts/noto-fonts/blob/115d38430d957d38307457c036302b7bdbe0bbc4/phaseIII_only/unhinted/otf/NotoSansMyanmar/NotoSansMyanmar-Regular.otf
Site: https://github.com/googlefonts/noto-fonts/blob/115d38430d957d38307457c036302b7bdbe0bbc4/unhinted/NotoSerifMyanmar/NotoSerifMyanmar-Regular.ttf
Date: 2020-04-14

### Font version
Version 2.001

### Issue
When a narrow consonant has two subjoined consonants and a medial <i>ra</i>, the second subjoined consonant consonant is drawn over the first. If the base consonant is wide or there is no medial <i>ra</i>, the subjoined consonants are stacked correctly.

### Character data
ဂ္တ္ဏြ
U+1002 MYANMAR LETTER GA
U+1039 MYANMAR SIGN VIRAMA
U+1010 MYANMAR LETTER TA
U+1039 MYANMAR SIGN VIRAMA
U+100F MYANMAR LETTER NNA
U+103C MYANMAR CONSONANT SIGN MEDIAL RA

### Screenshot
![ဂ္တ္ဏြ](https://user-images.githubusercontent.com/1124347/79348605-35a72f80-7f03-11ea-9e3f-79e779e96a97.png)
![ဂ္တ္ဏြ](https://user-images.githubusercontent.com/1124347/79348607-363fc600-7f03-11ea-94e8-1e603c6be3fc.png)

**Comment by nizarsq** [comment]
@dscorbett I have tried to render the sequence you provided using multiple 3rd parties fonts and all of them show this kind of overlap. I'm not an expert in Myanmar script,  but based on little research, the combination you provided does not exist anywhere. I'm wondering if this is a valid combination, in other words, is there a real word that has this combination? Thanks

**Comment by patchew** [comment]
This is unlikely to be found in any extant text.
Most stacked combinations are cognate from Pali-Sanskrit; <-gtṇr-> would be
an odd combination.

The counter to "is this a real word" is often:
— other analogous combinations work; only this one doesn't (?);
— relying solely on extant texts is dependent on access and knowledge,
where "what if we just haven't yet found THE manuscript that has this
sequence?" becomes the rallying cry.

It seems, however, that 4-element consonant clusters are less
well-supported.

**Comment by dscorbett** [comment]
In the Sanskrit word <i lang=sa>matupstriyā</i>, the cluster <i lang=sa>pstri</i> is written ⟨<span lang=sa>ပ္သ္တြိ</span>⟩ <U+1015, U+1039, U+101E, U+1039, U+1010, U+103C, U+102D> in Myanmar script. In Noto Sans Myanmar, that cluster is subject to this bug.
![ပ္သ္တြိ](https://user-images.githubusercontent.com/1124347/94074917-30121780-fdc8-11ea-883b-23940c1be5e1.png)
(I still think bugs should be fixed even without attestations: they may be low priority, but they are still bugs.)

**Comment by ohbendy** [comment]
@dscorbett How did you find these combinations? My Burmese Sanskrit conjunct tables don't have any four-letter consonant clusters.

I'd agree that this should be rectified. Although there are only a handful of Burmese Sanskritists who would ever have a use for clusters like this, and considering the need to somewhere draw a line on what's practical to implement, this case is generalisable to any cluster with three consonants and a medial Ra, so a fix would be systematic enough to cover all cases.

**Comment by dscorbett** [comment]
I found the original cluster, ⟨ဂ္တ္ဏြ⟩, by analyzing the GSUB table. It was intended to be a minimal reproducible test case for a bug, not a claim of linguistic attestation for that specific cluster. I found the second cluster, ⟨ပ္သ္တြိ⟩, by searching Sanskrit corpora until I found a cluster that when transliterated would reproduce the same bug.

**Comment by ohbendy** [comment]
Right. Normally we don't aim to support sequences that don't really occur, especially in a script like Burmese where the effort required quickly becomes exponential. If the Sanskrit corpora are public, could you post a link? I'd like to scrape any other clusters from it.

**Comment by dscorbett** [comment]
I found it in [GRETIL’s Sanskrit corpus](http://gretil.sub.uni-goettingen.de/gretil.html#cdl). I also looked in [OliverHellwig/sanskrit](https://github.com/OliverHellwig/sanskrit) but didn’t find any clusters relevant to this issue.

**Comment by ohbendy** [comment]
Thank you :)

**Comment by ohbendy** [comment]
In case of interest: https://github.com/ohbendy/Myanmar-font-resources/blob/master/Burmese%20Sanskrit%20conjuncts.txt
